src/client/components/builder/logic/ExpressionInput.tsx

```diff
@@ -51,9 +52,48 @@ export const ExpressionInput: FC<ExpressionInputProps> = ({
   ...props
 }) => {
   const [inputValue, setInputValue] = useState<string>(value)
+  const [showCalcBar, setShowCalcBar] = useState(false)
   const [selection, setSelection] = useState<{ start: number; end: number }>()
 
   const inputRef = useRef<HTMLInputElement>(null)
+  const wrapperRef = useRef<HTMLDivElement>(null)
+
+  // syncs input value with subsequent value updates
```

Comment:
The code in lines 61-80 is duplicated from line 98 onwards

Reply:
Thanks for catching this! I must've been tired when I was resolving merge conflicts with develop...